src/base/datetime/calendar/body/index.ts

```diff
@@ -0,0 +1,294 @@
+  CustomEventDetail,
+  dispatchCustomEvent
+} from "../../../kuc-base";
+import { en, zh, ja } from "../../resource/locale";
+import { getDisplayingDates } from "../../utils/index";
+
+export class BaseDateTimeCalendarBody extends KucBase {
+  @property({ type: Number }) month = 0;
+  @property({ type: Number }) year = 2021;
+  @property({ type: String }) language = "en";
+  @property({ type: String }) value = "";
+
+  private _locale = en;
+
+  update(changedProperties: PropertyValues) {
+    changedProperties.forEach((_oldValue, propName) => {
+      propName === "language" &&
+        (this._locale = this._getLocale(this.language));
+    });
+    super.update(changedProperties);
+  }
+
+  render() {
+    return html`
+      ${this._getStyleTagTemplate()}
+      <table class="kuc-base-datetime-calendar-body__table" role="grid">
+        ${this._getHeaderItemsTemplate()}<!--
+        -->${this._getDateItemsTemplate()}
+      </table>
+    `;
+  }
+
+  private _getHeaderItemsTemplate() {
+    return html`
+      <thead>
+        ${this._locale.WEEK_DAYS.map(wday => {
+          return html`
+            <th
+              class="kuc-base-datetime-calendar-body__header"
+              role="columnheader"
+              abbr="${wday.abbr}"
+            >
+              ${wday.text}
+            </th>
+          `;
+        })}
+      </thead>
+    `;
+  }
+
+  private _getDateItemsTemplate() {
+    const displayingDates = getDisplayingDates(this.year, this.month);
+    const today = this._getDateString();
+    return html`
+      <tbody>
+        ${displayingDates.map(weeks => {
+          return html`
+            <tr>
+              ${weeks.map(weekDate => {
+                const dateParts = weekDate.split("-");
+                return html`
+                  <td role="gridcell">
+                    <button
+                      aria-selected="${this.value === weekDate}"
+                      tabindex="${weekDate === today || this.value === weekDate
+                        ? "0"
+                        : "-1"}"
+                      class="kuc-base-datetime-calendar-body__date${this._getDateClass(
+                        dateParts
+                      )}"
+                      data-date="${weekDate}"
+                      @click=${this._handleClickDateCalendarBody}
+                      @keydown="${this._handleKeyDownDateCalendarBody}"
+                    >
+                      ${dateParts[2] || ""}
+                    </button>
+                  </td>
+                `;
+              })}
+            </tr>
+          `;
+        })}
+      </tbody>
+    `;
+  }
+
+  private _handleClickDateCalendarBody(event: MouseEvent | KeyboardEvent) {
+    event.preventDefault();
+    event.stopPropagation();
+
+    const itemEl = event.target as HTMLButtonElement;
+    itemEl.setAttribute("aria-selected", "true");
+
+    const value = itemEl.getAttribute("data-date") || "";
+    this._handleDispatchEvent(value);
+    this._close();
```

Comment:
@nmanhit 
It will make the body disapear when clicking.
Please remove it.

It will show/hide following Calendar (parent) part.